tests/models/test_hooks.py

```diff
@@ -553,7 +554,6 @@ def training_step(self, batch, batch_idx):
         dict(name="training_epoch_end", args=([dict(loss=ANY)] * train_batches,)),
         dict(name="Callback.on_train_epoch_end", args=(trainer, model)),
         # `ModelCheckpoint.save_checkpoint` is called here from `Callback.on_train_epoch_end`
-        dict(name="Callback.state_dict"),
```

Comment:
@mathemusician can you revert all changes to this test `test_trainer_model_hook_system_fit`? It is still failing